nomad/job_endpoint.go

```diff
@@ -433,6 +438,53 @@ func (j *Job) Register(args *structs.JobRegisterRequest, reply *structs.JobRegis
 	return nil
 }
 
+// specChanged checks to see if the job spec has changed. If the job is multiregion,
+// it checks all regions. This handles the case of a multiregion job that has been
+// stopped in one or more regions being restarted via the run command.
```

Comment:
The doc strings here says it's "handling" this case but it's a "what not why" comment: we should be explaining why checking that the spec has changed is meaningful and why we need to specially handle this case at all.